lib/components/discover/discover_body.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter/material.dart';
+import 'package:project_fomo/blocs/discover_bloc.dart';
+import 'package:project_fomo/components/discover/event_scroller.dart';
+import 'package:project_fomo/components/discover/featured_carousel.dart';
+import 'package:project_fomo/components/shared/page_header.dart';
+import 'package:project_fomo/models/event.dart';
+import 'package:provider/provider.dart';
+
+const double leftTitlePadding = 15.0;
+
+class DiscoverBody extends StatefulWidget {
+  @override
+  _DiscoverBodyState createState() => _DiscoverBodyState();
+}
+
+class _DiscoverBodyState extends State<DiscoverBody> {
+  DiscoverBloc _bloc;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _bloc = Provider.of<DiscoverBloc>(context, listen: false);
+  }
+
+  Widget build(BuildContext context) {
+    _bloc.refreshEventCategories();
```

Comment:
This does not need to change in our case because this page is kept alive, but normally you would need to move this call out of build so it does not get called on every rebuild.

Reply:
Good catch. This call has been moved to the didChangeDependencies() method.